custom_components/entity_controller/__init__.py

```diff
@@ -801,8 +807,14 @@ def turn_off_control_entities(self):
                 self.call_service(e, 'turn_on')
         else:
             for e in self.controlEntities:
+
                 self.log.debug("Turning off %s", e)
-                self.call_service(e, 'turn_off')
+
+                if self.offParams is not None:
```

Comment:
That's true, but we would still have to check `if self.lightParams.get('service_data_off') is not None`.

Reply:
For `service_data` you also check if it's not None. Because when you set `lightParams` you get the info from `service_data` and if there's no config set then it should be None. I used the same logic for `service_data_off`